runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -276,9 +276,6 @@ impl Wasmer2VM {
         let prepared_code = prepare::prepare_contract(code.code(), &self.config)
             .map_err(CompilationError::PrepareError)?;
 
-        self.engine
-            .validate(&prepared_code)
-            .map_err(|e| CompilationError::WasmerCompileError { msg: e.to_string() })?;
```

Comment:
Let's turn this into a debug assert explaining why we believe we don't need extra validation here?